lib/reservation_mgr.rb

```diff
@@ -0,0 +1,111 @@
+require 'pry'
+require 'date'
+require_relative 'room'
+require_relative 'reservation'
+
+class ReservationMgr
+  attr_reader :rooms , :reservations
+  def initialize(num_rooms)
+    @rooms = Room.create(num_rooms)
+    @reservations = []
+
+  end
+
+  def available_rooms(check_in,check_out,rooms: 1)
+    if check_out < check_in
+      raise StandardError.new('The check-out date is before the check-in')
+    end
+
+    available_rooms = []
+    @rooms.each do |room|
```

Comment:
You could also do `@rooms.select` to do the same thing here.

Reply:
Thanks! I didn't even think about using the is_available? functionality with the select enumerable!